src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -34,11 +45,157 @@ ucs_config_field_t uct_gga_mlx5_iface_config_table[] = {
   {NULL}
 };
 
+static ucs_status_t
+uct_ib_mlx5_gga_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
+                          void *address, size_t length,
+                          const uct_md_mkey_pack_params_t *params,
+                          void *mkey_buffer)
+{
+    uct_md_mkey_pack_params_t gga_params = *params;
+
+    if (gga_params.field_mask & UCT_MD_MKEY_PACK_FIELD_FLAGS) {
+        gga_params.flags      |= UCT_MD_MKEY_PACK_FLAG_EXPORT;
+    } else {
+        gga_params.field_mask |= UCT_MD_MKEY_PACK_FIELD_FLAGS;
+        gga_params.flags       = UCT_MD_MKEY_PACK_FLAG_EXPORT;
+    }
 
-typedef struct {
-    uct_rc_mlx5_iface_common_t  super;
-} uct_gga_mlx5_iface_t;
+    return uct_ib_mlx5_devx_mkey_pack(uct_md, uct_memh, address, length,
+                                      &gga_params, mkey_buffer);
+}
+
+static ucs_status_t
+uct_gga_mlx5_rkey_unpack(uct_component_t *component, const void *rkey_buffer,
+                         uct_rkey_t *rkey_p, void **handle_p)
+{
+    const uct_ib_md_packed_mkey_t *mkey = rkey_buffer;
+    uct_rkey_bundle_t *rkey_bundle;
+    uct_gga_mlx5_rkey_handle_t *rkey_handle;
+
+    rkey_bundle = ucs_malloc(sizeof(*rkey_bundle), "gga_rkey_bundle");
+    if (rkey_bundle == NULL) {
+        return UCS_ERR_NO_MEMORY;
+    }
```

Comment:
can we avoid allocating uct_rkey_bundle_t?

Reply:
using mpool?